src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -159,6 +188,316 @@ private string GetRootPathWithGuid_NoLock()
+                ? Wrapping.WrapAlways
+                : Wrapping.WrapIfTooLong;
+            formattingOptions.ArrayInitializerBraceStyle = GetBraceStyle(options.GetOption(CSharpFormattingOptions.NewLinesForBracesInObjectCollectionArrayInitializers));
+            formattingOptions.ChainedMethodCallWrapping = Wrapping.DoNotChange;
+            formattingOptions.MethodCallArgumentWrapping = Wrapping.DoNotChange;
+            formattingOptions.NewLineAferMethodCallOpenParentheses = NewLinePlacement.DoNotCare;
+            formattingOptions.MethodCallClosingParenthesesOnNewLine = NewLinePlacement.SameLine;
+            formattingOptions.IndexerArgumentWrapping = Wrapping.DoNotChange;
+            formattingOptions.NewLineAferIndexerOpenBracket = NewLinePlacement.DoNotCare;
+            formattingOptions.IndexerClosingBracketOnNewLine = NewLinePlacement.SameLine;
+            formattingOptions.MethodDeclarationParameterWrapping = Wrapping.DoNotChange;
+            formattingOptions.NewLineAferMethodDeclarationOpenParentheses = NewLinePlacement.DoNotCare;
+            formattingOptions.MethodDeclarationClosingParenthesesOnNewLine = NewLinePlacement.SameLine;
+            formattingOptions.IndexerDeclarationParameterWrapping = Wrapping.DoNotChange;
+            formattingOptions.NewLineAferIndexerDeclarationOpenBracket = NewLinePlacement.DoNotCare;
+            formattingOptions.IndexerDeclarationClosingBracketOnNewLine = NewLinePlacement.SameLine;
+            formattingOptions.AlignToFirstIndexerArgument = false;
+            formattingOptions.AlignToFirstIndexerDeclarationParameter = false;
+            formattingOptions.AlignToFirstMethodCallArgument = false;
+            formattingOptions.AlignToFirstMethodDeclarationParameter = false;
+            formattingOptions.NewLineBeforeNewQueryClause = NewLinePlacement.NewLine;
+
+            // Using declarations
+            formattingOptions.UsingPlacement = UsingPlacement.TopOfFile;
+
+            return decompilerSettings;
+        }
+
+        private class RoslynAssemblyResolver : IAssemblyResolver
+        {
+            private readonly Compilation parentCompilation;
+
+            public RoslynAssemblyResolver(Compilation parentCompilation)
+            {
+                this.parentCompilation = parentCompilation;
+            }
+
+            public AssemblyDefinition Resolve(AssemblyNameReference name)
+            {
+                return Resolve(name, new ReaderParameters());
+            }
+
+            public AssemblyDefinition Resolve(AssemblyNameReference name, ReaderParameters parameters)
+            {
+                foreach (var assembly in parentCompilation.GetReferencedAssemblySymbols())
+                {
+                    if (assembly.Identity.Name == name.Name
+                        && assembly.Identity.Version == name.Version
```

Comment:
&& on previous line at end of line to match IDE layer formatting.

Reply:
➡️ I'm hoping to work our way the other direction over time. As long as an expression is consistent with itself I'm happy.

Reply:
My personal preference would be to do that in one swell foop.  :)   But i know i'm inconsistent here as well, adopting some new patterns piecemeal before applying globally.

Reply:
Given we have an established style (&& on previous line) let's stick with that for now.